File: rust/01.rs
fn num1() -> u32 {
  let mut sum = 0;

  for n in 0..1000 {
    if n % 3 == 0 || n % 5 == 0 {
      sum += n;
    }
  }

  return sum;
}

fn num2() -> u32 {
  (0..1000).fold(0, |acc, val| if val % 3 == 0 || val % 5 == 0 {acc + val} else {acc})
}

fn main() {
  println!("Result: {}", num1());
  println!("Result: {}", num2());
}

File: clojure/01.clj
; one
(defn num1 []
  (reduce (fn [acc val]
    (if (or (= (mod val 3) 0) (= (mod val 5) 0))
      (+ acc val)
      acc))
    (range 1000)))

; two
(defn reducerfn [acc, val]
  (if (or (zero? (mod val 3)) (zero? (mod val 5)))
    (+ acc val)
    acc))

(defn num2 []
  (reduce reducerfn (range 1000)))

(defn -main
  []
  (println (num1))
  (println (num2)))
